libraries/dotnet/EnergyOriginEventStore/EnergyOriginEventStore.Tests/MemoryEventStoreTests.cs

```diff
@@ -0,0 +1,145 @@
+using EnergyOriginEventStore.EventStore;
+using EnergyOriginEventStore.EventStore.Memory;
+using EnergyOriginEventStore.Tests.Topics;
+using Xunit;
+
+namespace EnergyOriginEventStore.Tests;
+
+public class MemoryEventStoreTests
```

Comment:
We should add a test that performs the following steps:

1. Creates consumer and ensures that it is drained (using await Thread.Delay)
2. Produces a new message
3. Asserts that the consumer created in step 1 receives the message produced in step 2.